# Развитие игровой индустрии с 2000 по 2013 год

- Автор:Манузин Никита
- Дата:04.10.2025

### Цели и задачи проекта

<font color='#777778'>Используя датасет `/datasets/new_games.csv` сделаем обзор игровых платформ, изучим объёмы продаж игр разных жанров и региональные предпочтения игроков. Акцент сделаем на играх жанра RPG</font>

### Описание данных

<font color='#777778'>В проекте будут использованы данные `/datasets/new_games.csv` со следующим описанием:  
`Name` — название игры;  
`Platform` — название платформы;  
`Year of Release` — год выпуска игры;  
`Genre` — жанр игры;  
`NA sales` — продажи в Северной Америке (в миллионах проданных копий);  
`EU sales` — продажи в Европе (в миллионах проданных копий);  
`JP sales` — продажи в Японии (в миллионах проданных копий);  
`Other sales` — продажи в других странах (в миллионах проданных копий);  
`Critic Score` — оценка критиков (от 0 до 100);  
`User Score` — оценка пользователей (от 0 до 10);  
`Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.</font>

### Содержимое проекта

<font color='#777778'>1. Загрузка и знакомство с данными  
2. Проверка ошибок в данных и их предобработка  
3. Фильтрация данных  
4. Категоризация данных  </font>

---

## 1. Загрузка данных и знакомство с ними

- Загрузите необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.


In [1]:
# Пустые ячейки после каждого задания — примерное пространство для работы.
# Вы можете свободно добавлять или удалять ячейки по своему усмотрению в зависимости от логики и объёма работы.

In [2]:
# Используйте ячейки типа Code для вашего кода

In [3]:
# При необходимости добавьте новые ячейки для кода

In [4]:
import pandas as pd

In [5]:
df = pd.read_csv('/datasets/new_games.csv')

- Познакомьтесь с данными: выведите первые строки и результат метода `info()`.


In [6]:
display(df.head(10))

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN
5,Tetris,GB,1989.0,Puzzle,23.20,2.26,4.22,0.58,NaN,NaN,NaN
6,New Super Mario Bros.,DS,2006.0,Platform,11.28,9.14,6.5,2.88,89.0,8.5,E
7,Wii Play,Wii,2006.0,Misc,13.96,9.18,2.93,2.84,58.0,6.6,E
8,New Super Mario Bros. Wii,Wii,2009.0,Platform,14.44,6.94,4.7,2.24,87.0,8.4,E
9,Duck Hunt,NES,1984.0,Shooter,26.93,0.63,0.28,0.47,NaN,NaN,NaN


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


- Сделайте вывод о полученных данных: данные какого объёма вам предоставили, соответствуют ли они описанию, встречаются ли в них пропуски, используются ли верные типы данных.
- Отметьте другие особенности данных, которые вы обнаружили и на которые стоит обратить внимание при предобработке. Например, вы можете проверить названия столбцов: все ли названия отражают содержимое данных и прописаны в удобном для работы виде.

Датасет `/datasets/new_games.csv` содержит 11 столбцов и 16956 строк, в которорых представлена информация  о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр. Данные соответствуют представленному описанию. Рассмотрим данные подробнее:
- В столбцах `Name`, `Year of Release`, `Genre`, `Critic Score`, `User score` `Rating` присутствуют пропуски;  
- В столбце `Year of Release` используется тип данных `float64`. Этот столбец хранит информацию о годе выпуска игры, для таких данных более оптимальным будет тип `int64`, также можно уменьшить разрядность до int32 или int16, если значения достаточно малы и это не повлияет на точность;  
- Столбцы `EU sales` и `JP sales` имеют тип данных `object`, что не является оптимальным выбором, корректнее будет использовать тип данных `float64` по аналогии со столбцами `NA sales` и `Other sales`;  
- Столбец `User Score` хранит информацию об оценке пользователей(от 0 до 10), имеет тип `object`, так как представленные данные имеют дробную часть оптимальным типом данных будет являться тир `float64`;  
- В остальных столбцах тип данных использован верно, в преобразовании не нуждается.

Названия столбцов корректно отражают содержимое данных, однако прописаны в неудобном для работы виде. Потребуется преобразование к стилю snake case и приведение к нижнему регистру.

---

## 2.  Проверка ошибок в данных и их предобработка


### 2.1. Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [8]:
display(df.columns)

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [9]:
df.columns = df.columns.str.lower().str.replace(' ','_')

In [10]:
display(df.columns)

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')

### 2.2. Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

- Причиной некорректного типа данных в столбце `year_of_release` может являться наличие пропусков;  
- Причиной некорректного типа данных в столбцах `eu_sales` и `jp_sales` является наличие строковых значений 'unknown', а в столбце `user_score` наличие строкового значения 'tbd'.

 - В числовых столбцах могут встретиться строковые значения, например unknown или другие. Приводите такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

Преобразуем тип данных в столбцах `eu_sales`, `jp_sales` и `user_score` к числовому типу данных `float64`:

In [11]:
df['eu_sales'] = pd.to_numeric(df['eu_sales'], errors = 'coerce', downcast = 'float')

In [12]:
df['jp_sales'] = pd.to_numeric(df['jp_sales'], errors = 'coerce', downcast = 'float')

In [13]:
df['user_score'] = pd.to_numeric(df['user_score'], errors = 'coerce', downcast = 'float')

Аналогичным образом подберем разряд для данных в столбце `na_sales`:

In [14]:
df['na_sales'] = pd.to_numeric(df['na_sales'], downcast = 'float')

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16954 non-null  object 
 1   platform         16956 non-null  object 
 2   year_of_release  16681 non-null  float64
 3   genre            16954 non-null  object 
 4   na_sales         16956 non-null  float32
 5   eu_sales         16950 non-null  float32
 6   jp_sales         16952 non-null  float32
 7   other_sales      16956 non-null  float64
 8   critic_score     8242 non-null   float64
 9   user_score       7688 non-null   float32
 10  rating           10085 non-null  object 
dtypes: float32(4), float64(3), object(4)
memory usage: 1.2+ MB


### 2.3. Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


Посчитаем общее количество пропусков в каждом столбце:

In [16]:
df.isna().sum()

name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

Теперь оценим долю пропусков в каждом столбце:

In [17]:
df.isna().sum() / len(df) * 100

name                0.011795
platform            0.000000
year_of_release     1.621845
genre               0.011795
na_sales            0.000000
eu_sales            0.035386
jp_sales            0.023590
other_sales         0.000000
critic_score       51.391838
user_score         54.659118
rating             40.522529
dtype: float64

- Изучите данные с пропущенными значениями. Напишите промежуточный вывод: для каких столбцов характерны пропуски и сколько их. Предположите, почему пропуски могли возникнуть. Укажите, какие действия с этими данными можно сделать и почему.


При изучении данных с пропущенными значениями, можно отметить что в столбцах `name` и `genre` одинаковое количество пропусков в количестве 2 штук, что может быть связано как со сбоем при выгрузке данных, так и указывать на их взаимосвязь, так как при отсутствии названия игры невозможно отпределить ее жанр. Эти данные можно удалить, анализировать в дальнейшем игру, не зная ее названия не выглядит логичным, да и доля очень мала в общем объеме данных.

Пропуски в столбцах `critic_score`, `user_score`, и `rating` могут быть связаны с низкими или даже в некоторых случаях нулевыми продажами этих игр. Можно предположить, что рейтинг играм присваивается исходя из определенного уровня популярности и продаж. 


Эти данные пригодятся нам для дальнейшего исследования, однако удалять их не будем, так как потеряем значительную часть данных.

Пропуски в столбце `year_of_release` могут быть как следствием человеческого фактора, так и возниконовением ошибки при выгрузке или формировании датасета.

Данные этого столбца необходимы нам для ислледования. Доля пропусков составляет 1,6 % и позволяет удалить нам пропуски без существенного влияния на результат, так и поступим.

Пропуски в столбцах `eu_sales` и `jp_sales` появились в результате преобразования типа данных в этих столбцах. Заменим пропуски на среднее значение в зависимости от названия платформы и года выхода игры.

- Обработайте пропущенные значения. Для каждого случая вы можете выбрать оптимальный, на ваш взгляд, вариант: заменить на определённое значение, оставить как есть или удалить.
- Если вы решите заменить пропуски на значение-индикатор, то убедитесь, что предложенное значение не может быть использовано в данных.
- Если вы нашли пропуски в данных с количеством проданных копий игры в том или ином регионе, их можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.

In [18]:
# Перед обработкой пропусков сохраним исходное количество строк
first_row_count = df.shape[0]

Заменим пропуски в данных с количеством проданных копий игры на среднее значение в зависимости от названия платформы и года выхода игры:

In [19]:
def sales_count(row):
    if pd.isna(row['eu_sales']):
        group = df[(df['platform'] == row['platform']) &
                   (df['year_of_release'] == row['year_of_release'])]
        return group['eu_sales'].mean()
    else:
        return row['eu_sales']

df['eu_sales'] = df.apply(sales_count, axis=1)

In [20]:
def sales_count(row):
    if pd.isna(row['jp_sales']):
        group = df[(df['platform'] == row['platform']) &
                   (df['year_of_release'] == row['year_of_release'])]
        return group['jp_sales'].mean()
    else:
        return row['jp_sales']

df['jp_sales'] = df.apply(sales_count, axis=1)

Удалим из датафрейма строки с пропусками в столбцах `name`, `genre` и `year_of_release`:

In [21]:
df = df.dropna(subset = ['name', 'genre', 'year_of_release'])

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16679 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16679 non-null  object 
 1   platform         16679 non-null  object 
 2   year_of_release  16679 non-null  float64
 3   genre            16679 non-null  object 
 4   na_sales         16679 non-null  float32
 5   eu_sales         16679 non-null  float64
 6   jp_sales         16679 non-null  float64
 7   other_sales      16679 non-null  float64
 8   critic_score     8085 non-null   float64
 9   user_score       7558 non-null   float32
 10  rating           9901 non-null   object 
dtypes: float32(2), float64(5), object(4)
memory usage: 1.4+ MB


После удаления пропущенных значений преобразуем тип данных в столбце `year_of_release`

In [23]:
df['year_of_release'] = pd.to_numeric(df['year_of_release'], downcast = 'integer')

### 2.4. Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

Изучим уникальные значения в столбцах `genre`, `platform` и `rating`:

In [49]:
for column in ['platform','year_of_release','genre','rating']:
    print(f'Уникальные значения в столбце {column}:')
    print(df[column].sort_values().unique())
    print()

Уникальные значения в столбце platform:
['2600' '3DO' '3DS' 'DC' 'DS' 'GB' 'GBA' 'GC' 'GEN' 'GG' 'N64' 'NES' 'NG'
 'PC' 'PCFX' 'PS' 'PS2' 'PS3' 'PS4' 'PSP' 'PSV' 'SAT' 'SCD' 'SNES' 'TG16'
 'WS' 'Wii' 'WiiU' 'X360' 'XB' 'XOne']

Уникальные значения в столбце year_of_release:
[1980 1981 1982 1983 1984 1985 1986 1987 1988 1989 1990 1991 1992 1993
 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005 2006 2007
 2008 2009 2010 2011 2012 2013 2014 2015 2016]

Уникальные значения в столбце genre:
['action' 'adventure' 'fighting' 'misc' 'platform' 'puzzle' 'racing'
 'role-playing' 'shooter' 'simulation' 'sports' 'strategy']

Уникальные значения в столбце rating:
['AO' 'E' 'E10+' 'EC' 'K-A' 'M' 'RP' 'T' nan]



После изучения видим, что в столбце `genre` встречаются неявнеы дубликаты, связанные с разным регистром написания. Приведем все данные в столбце `genre` к нижнему регистру:

In [25]:
df['genre'] = df['genre'].str.lower()

- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

In [26]:
df_sorted = df.sort_values(by = df.columns.tolist())

In [27]:
duplicates = df_sorted[df_sorted.duplicated(keep=False)]

In [28]:
display(duplicates)

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
15191,Beyblade Burst,3DS,2016,role-playing,0.00,0.00,0.03,0.00,NaN,NaN,NaN
15192,Beyblade Burst,3DS,2016,role-playing,0.00,0.00,0.03,0.00,NaN,NaN,NaN
15301,11eyes: CrossOver,X360,2009,adventure,0.00,0.00,0.02,0.00,NaN,NaN,NaN
15302,11eyes: CrossOver,X360,2009,adventure,0.00,0.00,0.02,0.00,NaN,NaN,NaN
4860,18 Wheeler: American Pro Trucker,PS2,2001,racing,0.20,0.15,0.00,0.05,61.0,5.7,E
...,...,...,...,...,...,...,...,...,...,...,...
2909,Yu-Gi-Oh! The Falsebound Kingdom,GC,2002,strategy,0.49,0.13,0.07,0.02,NaN,NaN,NaN
6695,Zoo Resort 3D,3DS,2011,simulation,0.11,0.09,0.03,0.02,NaN,NaN,E
6696,Zoo Resort 3D,3DS,2011,simulation,0.11,0.09,0.03,0.02,NaN,NaN,E
8156,Zumba Fitness Rush,X360,2012,sports,0.00,0.16,0.00,0.02,73.0,6.2,E10+


- Напишите промежуточный вывод: укажите количество найденных дубликатов и действия по их обработке.

В результате проведенной обработки было обнаружено 235 дубликатов, которые были удалены

In [29]:
# Определим количество строк до удаления дубликатов
initial_row_count = df.shape[0]

In [30]:
# Удаляем дубликаты
df_no_duplicates = df_sorted.drop_duplicates()

In [31]:
# Сохраняем количество строк после удаления дубликатов
final_row_count = df_no_duplicates.shape[0]

In [32]:
#Посчитаем количество удаленных строк
count_duplicated_rows = initial_row_count - final_row_count

In [33]:
display(f"Количество строк до удаления дубликатов: {initial_row_count}")
display(f"Количество строк после удаления дубликатов: {final_row_count}")
display(f"Количество удаленных строк: {count_duplicated_rows}")

'Количество строк до удаления дубликатов: 16679'

'Количество строк после удаления дубликатов: 16444'

'Количество удаленных строк: 235'

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

In [34]:
# Подсчитаем общее количество удаленных строк

deleted_rows_count = first_row_count - final_row_count

In [35]:
# Подсчитаем долю удаленных строк

deleted_rows_per = deleted_rows_count / first_row_count * 100

In [36]:
display(f"Всего удалено строк: {deleted_rows_count}")
display(f"Доля удаленных строк: {deleted_rows_per}")

'Всего удалено строк: 512'

'Доля удаленных строк: 3.019580089643784'

- После проведения предобработки данных напишите общий промежуточный вывод.

В результате знакомства с данными и их предобработкой были произведены следующие действия:
- Названия столбцов приведены к единому формату - нижний регистр и стиль написания snake case;
- Были заменены типы данных в следующих столбцах:
    - В столбце `year_of_release` тип данных `float64` заменен на `integer16`;  
    - в столбцах `eu_sales` и `jp_sales` тип данных `object` заменен на `float64`;
    - В столбце `user_score`  тип `object` заменени на `float32`  
    - Отдельно заметим, что для оптимизации работы для каждого столбца с числовыми данными подобрана оптимальная разрядность

В 8 из 11 столбцов были обнаружены пропущенные значения, максимальное количество пропусков присутствует в столбце `user_score` (9268 строк в абсолютном значении, что составляет 54,65 % от общего количества строк), при обработке пропусков были осуществлены следующие действия:
- Пропуски в данных с количеством проданных копий игры (столбцы `eu_sales` и `jp_sales`) заменены на среднее значение в зависимости от названия платформы и года выхода игры;
- Пропуски в столбцах `genre`, `name` и `year_of_release` были удалены

Данные были предобработаны от явных и неявных дубликатов:
 - Данные в столбце `genre` приведены к нижнему регистру
 - Удалено 253 явных дубликата

В результате обработки данных общее количество удаленных строк составило 512, что составляет около 3% от первоначального объема данных

---

## 3. Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [37]:
# Отберем данные с 2000 по 2013 год и сохраним в отдельный датафрейм
df_actual = df_no_duplicates[(df_no_duplicates['year_of_release'] >= 2000) & (df_no_duplicates['year_of_release'] <= 2013)]

In [38]:
df_no_duplicates.isna().sum() / len(df) * 100

name                0.000000
platform            0.000000
year_of_release     0.000000
genre               0.000000
na_sales            0.000000
eu_sales            0.000000
jp_sales            0.000000
other_sales         0.000000
critic_score       50.728461
user_score         53.846154
rating             40.026380
dtype: float64

In [39]:
display(df_actual)

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
3394,Frozen: Olaf's Quest,3DS,2013,platform,0.27,0.27,0.00,0.05,NaN,NaN,NaN
3906,Frozen: Olaf's Quest,DS,2013,platform,0.21,0.26,0.00,0.04,NaN,NaN,NaN
2478,Tales of Xillia 2,PS3,2012,role-playing,0.20,0.12,0.45,0.07,71.0,7.9,T
8460,.hack//G.U. Vol.1//Rebirth,PS2,2006,role-playing,0.00,0.00,0.17,0.00,NaN,NaN,NaN
7182,.hack//G.U. Vol.2//Reminisce,PS2,2006,role-playing,0.11,0.09,0.00,0.03,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...
16731,thinkSMART: Chess for Kids,DS,2011,misc,0.01,0.00,0.00,0.00,NaN,NaN,E
647,uDraw Studio,Wii,2010,misc,1.65,0.57,0.00,0.20,71.0,NaN,E
8397,uDraw Studio: Instant Artist,Wii,2011,misc,0.06,0.09,0.00,0.02,NaN,NaN,E
15836,uDraw Studio: Instant Artist,X360,2011,misc,0.01,0.01,0.00,0.00,54.0,5.7,E


---

## 4. Категоризация данных
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

Столбцы `user_score` и `critic_score` содержат большое количество пропусков (53,8 и 50,7 % соответственно), поскольку удаление данных приведет к значительной потере данных, а заполнение данных средним значением может значительно исказить картину (оценка игр одинаковых жанров и выпущенных на одинаковых платформах не всегда являются одинаковыми. Даже разные части одной и той же серии игр могут иметь сильно разные оценки) для категоризации данных выделим четвертую категорию `нет оценки`, а результаты будем оценивать среди игр, у которых имеются оценки критиков и игроков.

In [40]:
df_actual = df_actual.copy()

In [41]:
def categorize_user_score(row):
    if pd.isna(row['user_score']):
        return 'нет оценки'
    elif row['user_score'] >= 8:
        return 'высокая оценка'
    elif row['user_score'] >= 3:
        return 'средняя оценка'
    else:
        return 'низкая оценка'

df_actual['user_category'] = df_actual.apply(categorize_user_score, axis=1)

- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [42]:
def categorize_critic_score(row):
    if pd.isna(row['critic_score']):
        return 'нет оценки'
    elif row['critic_score'] >= 80:
        return 'высокая оценка'
    elif row['critic_score'] >= 30:
        return 'средняя оценка'
    else:
        return 'низкая оценка'

df_actual['critic_category'] = df_actual.apply(categorize_critic_score, axis=1)

In [43]:
display(df_actual.head())

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating,user_category,critic_category
3394,Frozen: Olaf's Quest,3DS,2013,platform,0.27,0.27,0.00,0.05,NaN,NaN,NaN,нет оценки,нет оценки
3906,Frozen: Olaf's Quest,DS,2013,platform,0.21,0.26,0.00,0.04,NaN,NaN,NaN,нет оценки,нет оценки
2478,Tales of Xillia 2,PS3,2012,role-playing,0.20,0.12,0.45,0.07,71.0,7.9,T,средняя оценка,средняя оценка
8460,.hack//G.U. Vol.1//Rebirth,PS2,2006,role-playing,0.00,0.00,0.17,0.00,NaN,NaN,NaN,нет оценки,нет оценки
7182,.hack//G.U. Vol.2//Reminisce,PS2,2006,role-playing,0.11,0.09,0.00,0.03,NaN,NaN,NaN,нет оценки,нет оценки


- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [44]:
user_category_count = df_actual['user_category'].value_counts()
display(user_category_count)

нет оценки        6298
средняя оценка    4081
высокая оценка    2286
низкая оценка      116
Name: user_category, dtype: int64

In [45]:
critic_category_count = df_actual['critic_category'].value_counts()
display(critic_category_count)

нет оценки        5612
средняя оценка    5422
высокая оценка    1692
низкая оценка       55
Name: critic_category, dtype: int64

- Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [46]:
top_platforms = df_actual.groupby('platform')['name'].count()


In [47]:
display(top_platforms.sort_values(ascending = False))

platform
PS2     2127
DS      2120
Wii     1275
PSP     1180
X360    1121
PS3     1087
GBA      811
XB       803
PC       766
GC       542
3DS      300
PS       274
PSV      134
WiiU      74
N64       70
DC        31
GB        27
XOne      19
PS4       16
WS         4
Name: name, dtype: int64

---

## 5. Итоговый вывод

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

В результате проведенного исследования получен срез данных, который имеет 13 столбцов и 12781 строк:  
- `name` — название игры;  
- `platform` — название платформы;  
- `year_of_release` — год выпуска игры;  
- `genre` — жанр игры;  
- `na_sales` — продажи в Северной Америке (в миллионах проданных копий);  
- `eu_sales` — продажи в Европе (в миллионах проданных копий);
- `jp_sales` — продажи в Японии (в миллионах проданных копий);  
- `other_sales` — продажи в других странах (в миллионах проданных копий);  
- `critic_score` — оценка критиков (от 0 до 100);  
- `user_score` — оценка пользователей (от 0 до 10);  
- `rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board);  
- `user_category` - категория игры на основе оценок пользователей;  
- `critic_category` - категория игры на основе оценок критиков;  

Анализируя результаты проведенного ислледования можно выделить следующие основные моменты: 
- За период с 2000 по 2013 год было выпущено 12781 игра различных жанров и для разных платформ;
- Наиболее популярными плаформами являются: PS2, DS, Wii, PSP И X360;
- Согласно проведенной категоризации большая часть игр (без учета игр у которых отсутствуют данные об оценках пользователей и критиков) оценивалась по среднему уровню и критиками и пользователями